File: TwentyFortyEight/TFEMove.m
//
//  TFEMove.m
//  TwentyFortyEight
//
//  Created by Joshua Caswell on 3/4/16.
//  Copyright © 2016 Josh Caswell. All rights reserved.
//

#import "TFEMove.h"

@implementation TFEMove

+ (instancetype)moveForNode:(TFENode *)node
                   toSquare:(NSUInteger)destination
                  combining:(BOOL)combining 
                   spawning:(BOOL)spawning
{
    return [[self alloc] initWithNode:node
                             toSquare:destination
                            combining:combining
                             spawning:spawning];
}

- (instancetype)initWithNode:(TFENode *)node
                    toSquare:(NSUInteger)destination
                   combining:(BOOL)combining
                    spawning:(BOOL)spawning
{
    self = [super init];
    if( !self ) return nil;
    
    _node = node;
    _destination = destination;
    _combining = combining;
    _spawning = spawning;
    
    return self;
}

@end

File: Common/Game Core/TFEMove.swift
//
//  TFEMove.swift
//  TwentyFortyEight
//
//  Created by Joshua Caswell on 10/3/16.
//  Copyright © 2016 Josh Caswell. All rights reserved.
//

struct TFEMove
{
    let node: TFENode
    let destination: Int
    let isCombination: Bool
    let isSpawn: Bool
    
    init(node: TFENode, destination: Int, isCombination: Bool = false, isSpawn: Bool = false)
    {
        self.node = node
        self.destination = destination
        self.isCombination = isCombination
        self.isSpawn = isSpawn
    }
}
